### PR TITLE
4th - Lollipop and Unittests

### PR DESCRIPTION
**Pull request by ede0m**

### CODE REVIEW

File: floodviz/static/js/hydrograph.js [hunk 1]
```diff
@@ -10,39 +10,39 @@ document.addEventListener("DOMContentLoaded", function(event) {
...
-					.axisLabelDistance(10)
-					.ticks(5)
-					.tickFormat(function(d){return d3.time.format('%m-%d-%y')(new Date(d))});
-
-			chart.yAxis.axisLabel('Discharge (cubic feet per second)')
-					.axisLabelDistance(40)
-					//.ticks(5)
-					.tickValues([10, 100, 1000, 10000, 100000])
-					.tickFormat(function(d) { return d3.format(",")(d) + " cfps"});
+		// 	// might want to switch to lineChart(). When cumulative line chart is clicked, the Y axis freaks out and reverts back to issue at https://github.com/novus/nvd3/issues/695 #
```
Comment (mbucknell) [inline floodviz/static/js/hydrograph.js]: Before this is merged, this needs to be put back the way it was, realizing that this causes the code not to run.

File: floodviz/static/js/peakflow.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+	"use strict";
+
+	// Collect and set peakflow bar chart aspect ratio data
+	var peakflow_bar = document.getElementById('peakflow_bar');
+	peakflow_bar.style.height = FV.peakmeta['height'];
+	peakflow_bar.style.width = FV.peakmeta['width'];
+
+	var margin = {bottom: 100, right: 120, left: 120, top: 150};
+	var	width = parseInt(FV.peakmeta['width']);
```
Comment (mbucknell) [inline floodviz/static/js/peakflow.js]: Nit: Only one space after var.

File: floodviz/static/js/peakflow.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	var	height = parseInt(FV.peakmeta['height']);
+
+	var x = d3.scaleBand().rangeRound([0, width]).padding(.5);
+	var	y = d3.scaleLinear().range([height, 0]);
+
+	var xAxis = d3.axisBottom().scale(x);
+	var yAxis = d3.axisLeft().scale(y).ticks(8);
+
+	var svg = d3.select('#peakflow_bar').append('svg').attr("width", width + margin.left + margin.right)
```
Comment (mbucknell) [inline floodviz/static/js/peakflow.js]: Put each ```.function_name``` on a separate line

File: floodviz/static/js/peakflow.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+			.text("Year");
+		svg.append("g").attr('class', "axis axis--y").call(yAxis)
+			.append("text")
+			.attr("text-anchor", "middle")
+			.attr("transform", "rotate(-90)")
+			.attr("x", 0 - (height/2))
+			.attr("y", 0 - (margin.left/2))
+			.text("Discharge (cfps)");
+		
+		var lolli_data = data[data.length-1];
```
Comment (mbucknell) [inline floodviz/static/js/peakflow.js]: Add space around ```-```

File: floodviz/static/js/peakflow.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+			.text(title);
+
+		// create lollipop Stroke and Circle
+		var bars = d3.select('#peakflow_bar svg').selectAll('.bar')['_groups'][0];
+		var last_bar = bars[bars.length-1];
+		var penultimant_bar = bars[bars.length-2];
+		var lb_x = last_bar.x.baseVal.value;
+		var slb_x = penultimant_bar.x.baseVal.value;
+		var padding = lb_x - slb_x;
+		var lolli_pos_x = ((lb_x+padding+((1/2)*x.bandwidth())).toString())
```
Comment (mbucknell) [inline floodviz/static/js/peakflow.js]: Space around ```+``` and ```*```

File: floodviz/templates/index.html [hunk 6]
```diff
@@ -18,26 +16,27 @@
 		<div id="hydrograph">
 			<svg></svg>
 		</div>
+		<div id="peakflow_bar"></div>
 		<div id="map"></div>
```
Comment (mbucknell) [inline floodviz/templates/index.html]: I think we are going to want the map next to the hydrograph ultimately. Please put the peakflow-bar below the map

File: floodviz/templates/index.html [hunk 7]
```diff
@@ -18,26 +16,27 @@
...
 	</div>
 
 	<script type="text/javascript">
 		// Global Namespace for project - FV
 		var FV = FV || {};
-		FV.hydrometa = {{ config.HYDRO_META|safe }};
-		FV.mapinfo = {{ mapinfo | tojson |safe }};
```
Comment (mbucknell) [inline floodviz/templates/index.html]: Not sure why you are removing the mapinfo?

File: floodviz/templates/index.html [hunk 8]
```diff
@@ -18,26 +16,27 @@
...
 	</script>
 
 	<script type="text/javascript"
 			src="{{ url_for('static', filename='bower_components/jquery/dist/jquery.min.js') }}"></script>
 	<script type="text/javascript"
 			src="{{ url_for('static', filename='bower_components/bootstrap/dist/js/bootstrap.js') }}"></script>
 	<script type="text/javascript" src="{{ url_for('static', filename='bower_components/d3/d3.js') }}"></script>
-	<script type="text/javascript"
-			src="{{ url_for('static', filename='bower_components/nvd3/build/nv.d3.min.js') }}"></script>
+	<script type="text/javascript" src="{{ url_for('static', filename='js/hydrograph.js')}}"></script>
```
Comment (mbucknell) [inline floodviz/templates/index.html]: Should not include the hydrograph twice.

File: floodviz/views.py [hunk 9]
```diff
@@ -5,13 +5,18 @@
...
 from . import map_utils
+from . import peak_flow_utils
+
+url_nwis_prefix = app.config['NWIS_SITE_SERVICE_ENDPOINT']
 
 
 @app.route('/')
 def root():
     _hydrograph_helper()
-    mapinfo = _map_helper()
```
Comment (mbucknell) [inline floodviz/views.py]: Please add mapinfo back in.

File: floodviz/views.py [hunk 10]
```diff
@@ -27,25 +32,43 @@ def sitemap():
 
 
 def _hydrograph_helper():
-    # new data store #
+    # Hydrograph config vars #
+    sites = app.config['SITE_IDS']
```
Comment (mbucknell) [inline floodviz/views.py]: sites is in here twice.

File: floodviz/templates/index.html (hunk 7 above)
Comment (ede0m) [inline floodviz/templates/index.html]: This line was causing an error after pulling from master. I will put it back in.

File: floodviz/static/js/peakflow.js [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+	"use strict";
+
+	// Collect and set peakflow bar chart aspect ratio data
+	var peakflow_bar = document.getElementById('peakflow_bar');
+	peakflow_bar.style.height = FV.peakmeta['height'];
+	peakflow_bar.style.width = FV.peakmeta['width'];
+
+	var margin = {bottom: 100, right: 120, left: 120, top: 150};
+	var width = parseInt(FV.peakmeta['width']);
+	var	height = parseInt(FV.peakmeta['height']);
```
Comment (ede0m) [inline floodviz/static/js/peakflow.js]: Not sure why this is happening here. My text editor isn't showing this even after format indentation shortkey

File: floodviz/static/js/hydrograph.js [hunk 12]
```diff
@@ -42,7 +42,7 @@ document.addEventListener("DOMContentLoaded", function(event) {
 			nv.utils.windowResize(chart.update);
 			return chart;
 	
-		});
+		// });
```
Comment (ede0m) [inline floodviz/static/js/hydrograph.js]: Whoops! Missed this one!

**Pull request merged**